src/gluonts/dataset/tabular.py

```diff
@@ -0,0 +1,226 @@
+
+from typing import Iterator, List, Union
+from datetime import datetime
+
+import pandas as pd
+
+from gluonts.dataset.common import Dataset, DataEntry, ProcessDataEntry
+from gluonts.dataset.field_names import FieldName
+
+
+class DataFrameDataset(Dataset):
+    """
+    A pd.DataFrame-based dataset type.
+
+    This dataset uses the long format for each variable. Target time series values,
+    for example, are stacked on top of each other rather than side-by-side. The
+    same is true for other dynamic or categorical features. An example of a dataframe
+    is given below.
+
+
+    Parameters
+    ----------
+    data: pd.DataFrame
+        pd.DataFrame containing at least timestamp, target and item_id columns.
+    target: str or List[str]
+        Name of the column that contains the target time series.
+        For multivariate targets `target` is expecting a list of column names.
+    timestamp: str
+        Name of the column that contains the timestamp information.
+    item_id: str
+        Name of the column that, when grouped by, gives the different time series.
+    freq: str
+        Frequency of observations in the time series. Must be a valid pandas frequency.
+    feat_dynamic_real: List[str]
+        List of column names that contain dynamic real features.
+    feat_dynamic_cat: List[str]
+        List of column names that contain dynamic categorical features.
+    feat_static_real: List[str]
+        List of column names that contain static real features.
+    feat_static_cat: List[str]
+        List of column names that contain static categorical features.
+    past_feat_dynamic_real: List[str]
+        List of column names that contain dynamic real features only for the history.
+    prediction_length: int
+        For target and past dynamic features last `prediction_length` elements are
+        removed when iterating over the data set. This becomes important when the
+        predictor is called.
+    """
+
+    def __init__(
+        self,
+        data: pd.DataFrame,
+        target: str,
+        timestamp: str,
+        item_id: str,
+        freq: str,
+        feat_dynamic_real: List[str] = None,
+        feat_dynamic_cat: List[str] = None,
+        feat_static_real: List[str] = None,
+        feat_static_cat: List[str] = None,
+        past_feat_dynamic_real: List[str] = None,
+        prediction_length: int = None,
+    ) -> None:
+        self.data = data.copy()
+        self.target = target
+        self.timestamp = timestamp
+        self.item_id = item_id
+        self.freq = freq
+        self.prediction_length = prediction_length
+        self.feat_dynamic_real = feat_dynamic_real
+        self.feat_dynamic_cat = feat_dynamic_cat
+        self.feat_static_real = feat_static_real
+        self.feat_static_cat = feat_static_cat
+        self.past_feat_dynamic_real = past_feat_dynamic_real
+
+        self.one_dim_target = not isinstance(self.target, list)
+        self.process = ProcessDataEntry(
```

Comment:
I don't like how we do it in the other dataset classes, so I'm also not a fan having it here. Since it just does some type conversions (AFAIR), is there truly a need for this here, or could we just leverage pandas for this?

Reply:
The main reason, I use it, is to have the 'start' field processing in one place rather than making the DataFrameDataset to have its own logic for conversion. Meanwhile, it also does the conversions for other fields.
ProcessDataEntry is standardizing Dataset-outputs, which is quite nice. Whether to have the ProcessDataEntry as stadardizing mechanism or to implement a new one can maybe be discussed on another thread. What do you think?